renderer.php

```diff
@@ -71,6 +71,165 @@ public function render_ratingallocate_strategyform($mform) {
+                $buttondisteq->add_action(new confirm_action(
+                    get_string('distribute_unallocated_equally_confirm', RATINGALLOCATE_MOD_NAME)));
+
+                $distributeunallocatedurlfill = new moodle_url(
+                    $this->page->url,
+                    array('action' => ACTION_DISTRIBUTE_UNALLOCATED_FILL)
+                );
+                $buttondistfill = new single_button($distributeunallocatedurlfill,
+                    get_string('distributefill', RATINGALLOCATE_MOD_NAME), 'get');
+                $buttondistfill->add_action(new confirm_action(
+                    get_string('distribute_unallocated_fill_confirm', RATINGALLOCATE_MOD_NAME)));
+
+                // Add Amount of users that are unallocated and buttons to allocate them manually.
+                $this->add_table_row_triple($table,
+                    get_string('unallocated_user_count',
+                        RATINGALLOCATE_MOD_NAME,
+                        ['count' => $undistributeduserscount]) . $this->help_icon('distribution_description', RATINGALLOCATE_MOD_NAME),
+                    $this->render($buttondisteq),
+                    $this->render($buttondistfill)
+                );
+
+            } else if ($isdistributionrunning) {
+
+                // Add empty row.
+                $this->add_table_row_triple($table, '', '', '');
+
+                $this->add_table_row_triple($table,
+                    get_string('unallocated_user_count', RATINGALLOCATE_MOD_NAME, ['count' => $undistributeduserscount]),
+                    get_string('distribution_unallocated_already_running', RATINGALLOCATE_MOD_NAME),
+                    ''
+                );
+            }
+        }
+
+        $output .= html_writer::table($table);
+        $output .= $this->output->box_end();
+        $output .= $this->output->container_end();
+        return $output;
+    }
+
+    /**
+     * Displays the status concerning publishing the allocation together with the buttons to publish the allocation
+     * and to create groups.
+     *
+     * @param $ratingallocateid
+     * @param $coursemoduleid
+     * @param $status
+     * @return string
+     * @throws coding_exception
+     * @throws moodle_exception
+     */
+    public function render_ratingallocate_publish_allocation($ratingallocateid, $coursemoduleid, $status) {
+
+        $output = '';
+        $output .= $this->output->container_start('allocationstatustable');
+        $output .= $this->heading(get_string('publish_allocation_group', RATINGALLOCATE_MOD_NAME), 2);
+        $output .= $this->box_start();
+
+        $isready = $status === ratingallocate::DISTRIBUTION_STATUS_READY_ALLOC_STARTED;
+
+        $table = new html_table();
+
+        // Get description dependent on status.
+        $descriptionbaseid = 'publish_allocation_group_desc_';
+        $description = get_string($descriptionbaseid . $status, RATINGALLOCATE_MOD_NAME);
+
+        $this->add_table_row_triple($table,
```

Comment:
It would be nice if the table columns are aligned similar to the table of modify allocation so that all buttons are aligned the same. Maybe have a look at some css for this.